packages/jest-runtime/src/index.js

```diff
@@ -557,13 +612,35 @@ class Runtime {
     this._currentlyExecutingModulePath = lastExecutingModulePath;
   }
 
-  _requireCoreModule(moduleName: string) {
-    if (moduleName === 'process') {
-      return this._environment.global.process;
+  _requireNativeModule(
+    localModule: Module,
+    moduleName: string,
+    moduleRegistry: ModuleRegistry,
+    from: Path,
+  ) {
+    switch (moduleName) {
+      case 'buffer': // Causes issues when passing buffers to another context.
+      case 'module': // Calls into native_module, which is not mockable.
+      case 'os': // Pure native module.
```

Comment:
How do we maintain this whitelist?

Reply:
This way 😄 . There is no way of knowing that `os` is a native module, and that `module` cannot be re-evaluated. Essentially trial/error.

Reply:
Actually, `os` module should probably be wrapped with an object on top. Once we get that merged I will open a task for that.

Reply:
Ok, so I did a bunch of things in the last commit:
* I added a test suite that requires all native modules, so this will help us to semi-automate this list. 
* I also deep copy `os` and `v8` modules so that they are sandboxed as well (i.e. if someone modifies them, they won't leak).
* And finally I fixed the V8 crash for Node 8 (`async_hooks` can't be re-required).